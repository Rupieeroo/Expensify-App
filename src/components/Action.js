import React from 'react';

const Action = (props) => (
    <div>
      <button
        onClick={props.pickOptions}
        disabled={!props.hasOptions}
      >
        What Should I Do?
      </button>
    </div>
  );

export default Action;
