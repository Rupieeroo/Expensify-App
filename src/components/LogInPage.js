import React from 'react';
import { connect } from 'react-redux';
import { startLogIn } from '../actions/auth';

export const LogInPage = ({ startLogIn }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Manage your expenses, at no expense!</p>
      <button className="button" onClick={startLogIn}>Log In</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogIn: () => dispatch(startLogIn())
});

export default connect(undefined, mapDispatchToProps)(LogInPage);
