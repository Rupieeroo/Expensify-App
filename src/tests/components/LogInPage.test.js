import React from 'react';
import { shallow } from 'enzyme';
import { LogInPage } from '../../components/LogInPage';

test('should render LogInPage correctly', () => {
  const wrapper = shallow(<LogInPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogIn on button click', () => {
  const startLogIn = jest.fn();
  const wrapper = shallow(<LogInPage startLogIn={startLogIn}/>);
  wrapper.find('button').simulate('click');

  expect(startLogIn).toHaveBeenCalled();
});
