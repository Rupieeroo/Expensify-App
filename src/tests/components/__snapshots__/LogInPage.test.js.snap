// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LogInPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Manage your expenses, at no expense!
    </p>
    <button
      className="button"
    >
      Log In
    </button>
  </div>
</div>
`;
